Bottom: b3c1c46cb6dba963dc624d16f025a9f5d4067a57
Top:    be363292394095bbd8b408bcaea3b34e8acf845f
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-11-09 22:52:23 +0000

Refresh of play-with-glsl-shader-for-rend

---

diff --git a/src/hid/common/hidgl.c b/src/hid/common/hidgl.c
index 9f24340..38bb75e 100644
--- a/src/hid/common/hidgl.c
+++ b/src/hid/common/hidgl.c
@@ -146,14 +146,14 @@ hidgl_flush_triangles (triangle_buffer *buffer)
     buffer->triangle_array = NULL;
   }
 
-  glEnableClientState (GL_VERTEX_ARRAY);
   glVertexPointer (3, GL_FLOAT, 5 * sizeof (GLfloat), buffer->use_vbo ?
                      BUF_OFFSET (0) : buffer->triangle_array + 0);
 
-  glEnableClientState (GL_TEXTURE_COORD_ARRAY);
   glTexCoordPointer (2, GL_FLOAT, 5 * sizeof (GLfloat), buffer->use_vbo ?
                        BUF_OFFSET (3) : buffer->triangle_array + 3);
 
+  glEnableClientState (GL_VERTEX_ARRAY);
+  glEnableClientState (GL_TEXTURE_COORD_ARRAY);
   glDrawArrays (GL_TRIANGLE_STRIP, 0, buffer->vertex_count);
   glDisableClientState (GL_VERTEX_ARRAY);
   glDisableClientState (GL_TEXTURE_COORD_ARRAY);
